lib/logstash/outputs/cloudwatch.rb

```diff
@@ -166,28 +166,25 @@ def register
     @cw = Aws::CloudWatch::Client.new(aws_options_hash)
 
     @event_queue = SizedQueue.new(@queue_size)
-    @scheduler = Rufus::Scheduler.new
-    @job = @scheduler.schedule_every @timeframe do
-      @logger.debug("Scheduler Activated")
-      publish(aggregate({}))
-    end
+
+    @job = scheduler.every(@timeframe) { do_publish }
   end # def register
 
-  # Rufus::Scheduler >= 3.4 moved the Time impl into a gem EoTime = ::EtOrbi::EoTime`
-  # Rufus::Scheduler 3.1 - 3.3 using it's own Time impl `Rufus::Scheduler::ZoTime`
-  RufusTimeImpl = defined?(Rufus::Scheduler::EoTime) ? Rufus::Scheduler::EoTime :
-                      (defined?(Rufus::Scheduler::ZoTime) ? Rufus::Scheduler::ZoTime : ::Time)
+  def do_publish
+    publish(aggregate({}))
+  end
+  private :do_publish
 
   public
   def receive(event)
     return unless (event.get(@field_metricname) || @metricname)
 
     if (@event_queue.length >= @event_queue.max)
-      @job.trigger RufusTimeImpl.now
```

Comment:
we do not really need  to expose a `job#trigger` API,
the idea here is that we simply skip the job and execute the task on the same thread, when queue is full

Reply:
agree. given the size is full and the queue is blocked, running in the same thread doesn't hurt.